torch/distributed/fsdp/_optim_utils.py

```diff
@@ -340,7 +340,7 @@ def _broadcast_processed_state(
     if fsdp_state.rank == 0:
         objects[0] = tree_map_only(
             torch.Tensor,
-            lambda v: v.cpu() if v.dim() == 0 else _PosDimTensorInfo(v.shape, v.dtype),
+            lambda v: v.cpu() if v.dim() == 0 else _PosDimTensorInfo(v.shape, v.dtype),  # type: ignore[union-attr]
```

Comment:
this seems like a `mypy` bug; somehow it thinks `v` has a `Never` type (afaik that is just the uninhabited type.)